JuliaExperimental/gap/singular.g

```diff
@@ -295,15 +293,17 @@ InstallOtherMethod( GcdOp,
     [ "IsSingularPolynomialRing", "IsSingularPolynomial",
                                   "IsSingularPolynomial" ],
     function( R, f, g )
-        return SingularElement( f, Julia.Base.gcd( f, g ) );
+        return SingularElement( f, Julia.Base.gcd(
+                   JuliaPointer( f ), JuliaPointer( g ) ) );
     end );
 
 
 InstallOtherMethod( GcdOp,
     IsIdenticalObj,
     [ "IsSingularPolynomial", "IsSingularPolynomial" ],
     function( f, g )
-        return SingularElement( f, Julia.Base.gcd( f, g ) );
+        return SingularElement( f, Julia.Base.gcd(
+                   JuliaPointer( f ), JuliaPointer( g ) ) );
```

Comment:
@sebasguts and me just discussed an extension to the conversion code that would in effect automatically call `JuliaPointer` to "unwrap" high-level wrapper objects, as long as those high-level objects are in a suitable filter (say, `IsAutomaticUnwrappingJuliaWrapper` or so...). Then the above change could be reverted to the old version.

Reply:
Should this behavior (automatically unwrapping or not) really belong on an object,
or does it depend on the situation whether one wants to unwrap or not?

Reply:
We have been discussing this as part of how to deal with extended automatic conversions of arguments for method calls (as opposed to just performing the absolute necessary conversions). We think we have a good plan for that now, and are working on implementing and writing up more about this. In the meantime, this is nothing that should delay this PR.